packages/frontend/src/utils/twoFactor.js

```diff
@@ -42,6 +43,27 @@ export class TwoFactor extends Account2FA {
     }
 
     async initTwoFactor(accountId, method) {
+        // additional check if the ledger is enabled, in case the user was able to omit disabled buttons
```

Comment:
As much as I hate to say it, the new logic in `initTwoFactor` might be better as a new method on the wallet class. In this method all we need to know is `is there a ledger associated with this account?`, which relies on a multiple wallet methods and lots of wallet state but does not rely on the parameters to `initTwoFactor`.

From a clarity (and, optimistically, a future refactor) perspective, it flows a lot better if all the `initTwoFactor` method needs to be aware of is whether there's a ledger enabled - not how that determination is made. If `initTwoFactor` can simply call a wallet method then it doesn't need to take on this extra baggage, e.g.
```js
const isLedgerEnabled = await wallet.isLedgerEnabled();
if (!isLedgerEnabled) {
  throw new WalletError('Ledger Hardware Wallet is enabled', 'initTwoFactor.ledgerEnabled');
}
```

As an added bonus this new `isLedgerEnabled` method can be reused in other contexts in which we need to query whether a ledger is kinda-sorta-maybe enabled 😄